src/com/lftechnology/java/training/krishna/employeemanagement/service/employee/impl/EmployeeServiceImpl.java

```diff
@@ -0,0 +1,94 @@
+
+package com.lftechnology.java.training.krishna.employeemanagement.service.employee.impl;
+
+import java.util.List;
+
+import com.lftechnology.java.training.krishna.employeemanagement.dao.employee.impl.EmployeeDaoImpl;
+import com.lftechnology.java.training.krishna.employeemanagement.domain.Employee;
+import com.lftechnology.java.training.krishna.employeemanagement.service.employee.EmployeeService;
+import com.lftechnology.java.training.krishna.employeemanagement.utils.DateUtils;
+
+/**
+ * Implementation of employee service interface.
+ *
+ * @author Krishna Timilsina <krishnatimilsina@lftechnology.com>
+ * @version 1.0
+ */
+public class EmployeeServiceImpl implements EmployeeService {
+
+	private EmployeeDaoImpl employeeDaoImpl;
+
+	public EmployeeServiceImpl() {
+
+		this.employeeDaoImpl = new EmployeeDaoImpl();
+	}
+
+	@Override
+	public boolean delete(String fullname) {
+
+		Employee emp = new Employee();
+		emp.setFullname(fullname);
+		boolean isDeleted = this.employeeDaoImpl.delete(emp);
+		return isDeleted;
+	}
+
+	@Override
+	public List<Employee> findAll(String key) {
+
+		List<Employee> empList = this.employeeDaoImpl.findAll(key);
+		return empList;
+	}
+
+	@Override
+	public boolean terminate(String fullname) {
+
+		Employee emp = new Employee();
+		emp.setFullname(fullname);
+		emp.setModifiedAt(DateUtils.getCurrentTimeStamp());
+		boolean isTerminated = this.employeeDaoImpl.terminate(emp);
+		return isTerminated;
+	}
+
+	@Override
+	public Employee saveEmployee(
+		String username, String password, String fullname, String department,
+		String address, String role) {
+
+		Employee employee = null;
+		boolean isDuplicate = this.employeeDaoImpl.isDuplicate(username);
+		if (!isDuplicate) {
+			Employee emp = new Employee();
+			emp.setUsername(username);
+			emp.setPassword(password);
+			emp.setFullname(fullname);
+			emp.setDepartment(department);
+			emp.setAddress(address);
+			emp.setRole(role);
+			emp.setCreatedAt(DateUtils.getCurrentTimeStamp());
+			employee = this.employeeDaoImpl.create(emp);
+		}
+		return employee;
+	}
+
+	@Override
+	public Employee updateEmployee(
+		int id, String fullname, String department, String address) {
+
+		Employee emp = new Employee();
+		emp.setId(id);
+		emp.setFullname(fullname);
+		emp.setDepartment(department);
+		emp.setAddress(address);
+		emp.setModifiedAt(DateUtils.getCurrentTimeStamp());
+		Employee employee = this.employeeDaoImpl.update(emp);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Immediately return this expression instead of assigning it to the temporary variable "employee". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1488)